sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/NuGetMetadataReference.cs

```diff
@@ -84,22 +64,25 @@ private static IPackageRepository CreatePackageRepository()
 
             return aggregateRepository;
         }
-
-        private static MetadataReference[] CreateReferences(string packageId, string packageVersion)
+        
+        public static MetadataReference[] Create(string packageId, string packageVersion)
         {
-            var matchingDlls = Directory.GetFiles(GetNuGetPackageDirectory(), "*.dll", SearchOption.AllDirectories)
+            EnsurePackageIsInstalled(packageId, packageVersion);
```

Comment:
This is the main change - check the packages are available when required. This fixes the issue that caused the cix failure.